### PR TITLE
Chronos: fix thread num error when use `get_context` and `predict_with_onnx` together

### PR DESCRIPTION
**Pull request by plusbang**
## Description

When we call `forecaster.get_context(thread_num=4)` and `forecaster.predict_with_onnx`, we receive warning to set OMP_NUM_THREADS=1. Obviously, it's incorrect.
In this pr, fix it.

### 2. User API changes

None

### 4. How to test?
- [x] Unit test

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/pytorch/context_manager.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ def __enter__(self):
         # call `torch.set_num_threads` at most once
         self.forecaster.thread_num = set_pytorch_thread(self.thread_num,
                                                         self.forecaster.thread_num)
+        if hasattr(self.forecaster, 'optimized_model_thread_num'):
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/pytorch/context_manager.py:60]: In what case a forecaster does not has this `optimized_model_thread_num`? remind me please :)

**Review by TheaperDeng — COMMENTED** [review]
others LGTM

File: python/chronos/src/bigdl/chronos/pytorch/context_manager.py (hunk 1 above)
Comment (plusbang) [reply python/chronos/src/bigdl/chronos/pytorch/context_manager.py:60]: > In what case a forecaster does not has this `optimized_model_thread_num`? remind me please :)

yeah, autoformer forecaster now only has `thread_num`, and I think we don't need to add `optimized_model_thread_num` to it currently.

Comment (TheaperDeng) [reply python/chronos/src/bigdl/chronos/pytorch/context_manager.py:60]: ok make sense

**Review by TheaperDeng — APPROVED** [review]

**Pull request merged**